examples/tutorials/asr_inference_with_ctc_decoder_tutorial.py

```diff
@@ -0,0 +1,247 @@
+"""
+ASR Inference with CTC Decoder
+==============================
+
+**Author**: `Caroline Chen <carolinechen@fb.com>`__
+
+This tutorial shows how to perform speech recognition inference using a
+CTC beam search decoder with lexicon constraint and KenLM language model
+support. We demonstrate this on a pretrained wav2vec 2.0 model trained
+using CTC loss.
+
+"""
+
+
+######################################################################
+# Overview
+# --------
+# 
+# Running ASR inference using a CTC Beam Search decoder with a KenLM
+# language model and lexicon constraint requires the following components
+# 
+# -  Acoustic Model: model predicting phonetics from audio waveforms
+# -  Tokens: the possible predicted tokens from the acoustic model
+# -  Lexicon: mapping between possible words and their corresponding
+#    tokens sequence
+# -  KenLM: n-gram language model trained with the `KenLM
+#    library <https://kheafield.com/code/kenlm/>`__
+# 
+
+
+######################################################################
+# Preparation
+# -----------
+# 
+# First we import the necessary utilities and fetch the data that we are
+# working with
+# 
+
+import os
+
+import torch
+import torchaudio
+import requests
+import IPython
+
+
+######################################################################
+# Helper Functions
+# ~~~~~~~~~~~~~~~~
+# 
+
+def download_file(url, file):
```

Comment:
This might be coming from the other tutorials but I think we can simply use `torch.hub.download_url_to_file`.